src/view/observer.js

```diff
@@ -313,10 +313,13 @@ Candy.View.Observer = (function(self, $) {
 	 *
 	 * Parameters:
 	 *   (jQuery.Event) event - jQuery Event object
-	 *   (Object) args - {presetJid}
+	 *   (Object) args - {messageKey, presetJid}
 	 */
 	self.Login = function(event, args) {
-		Candy.View.Pane.Chat.Modal.showLoginForm(null, args.presetJid);
+		args = args || {};
+		args.messageKey = args.messageKey || '';
+		args.presetJid = args.presetJid || null;
```

Comment:
If args is undefined, args.XXX will cause an error. (XXX is anything, presetJid or messageKey in this case).

If I wanted to be insulted, I'd get married.  Use it or don't, I'm done.  Good luck with the code.